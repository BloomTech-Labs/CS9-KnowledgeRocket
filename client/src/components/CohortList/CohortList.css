.CohortListContainer {
	margin-left: 1rem;
	padding: 1rem;
	display: flex;
	flex-direction: row;
	align-items: flex-start;
	justify-content: space-around;
	width: 100%;
	border: 1px solid rgb(119, 136, 153);
	border-radius: 0.4rem;
}

.CohortCard {
	/*margin: 0 1rem 0 1rem;*/
}

.CohortCard_AddBtn {
	margin: 10px 0.5rem;
}
