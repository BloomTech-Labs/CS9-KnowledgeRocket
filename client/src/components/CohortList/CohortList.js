import React, { Component } from 'react';
import styled from 'styled-components';
import axios from 'axios';
import { connect } from 'react-redux';
import { generateBreadCrumbs } from '../../actions';
// Material Components
import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import Button from '@material-ui/core/Button';
import AddIcon from '@material-ui/icons/Add';
// Components
import CohortCard from '../CohortCard/CohortCard';

function mapStateToProps(state) {
    return {
        state,
    };
}

const CohortListContainer = styled.div`
  margin-left: 1rem;
  padding: 1rem;
  display: flex;
  flex-direction: row;
  align-items: flex-start;
  justify-content: space-around;
  width: 100%;
  border: 1px solid rgb(119, 136, 153);
  border-radius: 0.4rem;
`;

const CohortCardContainer = styled.div`
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  padding: 20px;
  width: 100%;
`;

const StyledCohortCard = styled(CohortCard)`
  margin: 20px;
`;

const AddButtonCard = styled(Card)`
  margin: 20px;
  width: 150px;
  text-align: center;
`;

const StyledCardContent = styled(CardContent)`
  margin: 10px 0.5rem;
`;

const AddButtonCardTitle = styled.h3`
  display: block;
  margin-bottom: 20px;
`;

// RENDERS A LIST OF COHORT CARDS
class CohortList extends Component {
  state = {
    cohort: [],
  };

  componentDidMount() {
    // Checks for User to be Authenticated
    // If not authenticated it will send the user to <login/>
    // If authenticated it will set the state with the current user.
    if (!this.props.state.user.authenticated) {
        this.props.history.push('/rocket/auth');
    }
    this.props.generateBreadCrumbs(this.props.history.location.pathname);
  }

  fetchCohortData = () => {
    // FETCH COHORT DATA FOR A USER FROM SERVER
    axios
      .get('http://localhost:5000/api/cohort')
      .then(response => {
        this.setState(() => ({ cohort: response.data }));
      })
      .catch(error => {
        console.error('Server Error', error);
      });
  };

  render() {
    return (
      <CohortListContainer>
        {this.state.cohort.length > 0 ? (
          // user has at least one cohort, render a cohort card
          <CohortCardContainer>
            {this.state.cohort.map((cohort, index) => (
              <StyledCohortCard key={`${cohort.students[index]}`} cohort={cohort} />
            ))}
            <AddButtonCard>
              <StyledCardContent>
                <AddButtonCardTitle>New Class</AddButtonCardTitle>
                <Button variant="fab" color="primary">
                  <AddIcon />
                </Button>
              </StyledCardContent>
            </AddButtonCard>
          </CohortCardContainer>
        ) : (
          // user has 0 cohorts, render add new class btn
          <AddButtonCard>
            <StyledCardContent>
              <AddButtonCardTitle>Add a new class</AddButtonCardTitle>
              <Button variant="fab" color="primary">
                <AddIcon />
              </Button>
            </StyledCardContent>
          </AddButtonCard>
        )}
      </CohortListContainer>
    );
  }
}

export default connect(mapStateToProps, { generateBreadCrumbs })(CohortList);
