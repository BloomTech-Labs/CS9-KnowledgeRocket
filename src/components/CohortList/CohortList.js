import React, { Component } from 'react';
import { connect } from 'react-redux';

function mapStateToProps(state) {
    return {

    };
}

class CohortList extends Component {
    render() {
        return (
            <div>
                
            </div>
        );
    }
}

export default connect(
    mapStateToProps,
)(CohortList);