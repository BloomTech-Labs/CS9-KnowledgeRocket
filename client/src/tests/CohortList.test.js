import React from 'react';
import renderer from 'react-test-renderer';
// Component
import CohortList from '../components/CohortList/CohortList';

describe('CohortList', () => {
	it('renders correctly', () => {
		const tree = renderer.create(<CohortList />).toJSON();
		expect(tree).toMatchSnapshot();
	});
});
