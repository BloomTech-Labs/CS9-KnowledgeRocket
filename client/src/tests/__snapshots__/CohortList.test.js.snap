// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CohortList renders correctly 1`] = `
<div
  className="sc-bwzfXH fliMDs"
>
  <div
    className="sc-htpNat jAgoUU"
  >
    <div
      className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1 sc-bxivhb LuvEN"
    >
      <div
        className="MuiCardContent-root-29"
      >
        <div
          className="MuiFormControl-root-30"
        >
          <div
            className="MuiCardContent-root-29"
          >
            <a
              className="sc-bdVaJa kKOtui"
              href="/rocket/classForm"
              onClick={[Function]}
            >
              
            </a>
          </div>
          <div
            className="MuiCardContent-root-29"
          >
            Students:
             
            0
          </div>
          <div
            className="MuiCardContent-root-29"
          >
            Participation:
          </div>
          <div
            className="MuiCardContent-root-29"
          >
            Rockets Sent:
          </div>
        </div>
      </div>
    </div>
    <div
      className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1 sc-ifAKCX jzUTef"
    >
      <div
        className="MuiCardContent-root-29 sc-EHOje fhnXmt"
      >
        <h3
          className="sc-bZQynM hyWVmf"
        >
          New Class
        </h3>
        <button
          className="MuiButtonBase-root-60 MuiButton-root-34 MuiButton-fab-51 MuiButton-contained-45 MuiButton-containedPrimary-46 MuiButton-raised-48 MuiButton-raisedPrimary-49"
          disabled={false}
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiButton-label-35"
          >
            <a
              className="sc-gzVnrw frFvIN"
              href="/rocket/classForm"
              onClick={[Function]}
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root-63"
                color={undefined}
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
                />
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
              </svg>
            </a>
          </span>
          <span
            className="MuiTouchRipple-root-70"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;
